Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -0,0 +1,65 @@
+require "tempfile"
+
+module Homebrew
+  module Diagnostic
+    class Checks
+      alias generic_check_tmpdir_sticky_bit check_tmpdir_sticky_bit
+
+      def check_tmpdir_sticky_bit
+        message = generic_check_tmpdir_sticky_bit
+        return if message.nil?
+        message + <<~EOS
+          If you don't have administrative privileges on this machine,
+          create a directory and set the HOMEBREW_TEMP environment variable,
+          for example:
+            export HOMEBREW_TEMP=~/tmp
```

Comment:
No need to `export` here.

Reply:
I'm afraid that I don't follow, Mike. If the `HOMEBREW_TEMP` shell variable were not exported to the environment, it would not be available to `brew`.

Reply:
Ah, I understand what you're saying. `export` is not needed to modify the shell profile. It is however needed for `brew` to work in this shell, which is the purpose of the `export`.

Reply:
Yes, that's the reason for having `export` in front of the first assignment (idea is that people can simply copy-paste these instructions). So, are we good here or should I remove this `export`?